addons/mail/static/src/messaging/component/activity_mark_done_popover/activity_mark_done_popover_tests.js

```diff
@@ -0,0 +1,285 @@
+        can_write: true,
+        forceNext: true,
+        id:12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        'Popover component should be present'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_feedback',
+        'Popover component should contain the feedback textarea'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_buttons',
+        'Popover component should contain the action buttons'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneScheduleNextButton',
+        'Popover component should contain the done & schedule next button'
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneButton',
+        'Popover component should NOT contain the done button'
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ActivityMarkDonePopover_discardButton',
+        'Popover component should NOT contain the discard button'
+    );
+});
+
+QUnit.test('activity mark done popover click on discard', async function (assert) {
+    assert.expect(4);
+
+    await this.start();
+    const activity = this.env.entities.Activity.create({
+        activity_category: 'not_upload_file',
+        can_write: true,
+        id:12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+    document.addEventListener('o-discard-clicked', () => {
+        assert.step('event_triggered');
+    }, true);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        'Popover component should be present'
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_discardButton',
+        'Popover component should contain the discard button'
+    );
+    document.querySelector('.o_ActivityMarkDonePopover_discardButton').click();
+    assert.verifySteps(['event_triggered'], 'Discard clicked should trigger the right event');
+
+});
+
+QUnit.test('activity mark done popover mark done without feedback', async function (assert) {
+    assert.expect(7);
+
+    await this.start({
+        debug: true,
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.activity/action_feedback') {
+                assert.step('action_feedback');
+                assert.strictEqual(args.args.length, 1);
+                assert.strictEqual(args.args[0].length, 1);
+                assert.strictEqual(args.args[0][0], 12);
+                assert.strictEqual(args.kwargs.attachment_ids.length, 0);
+                assert.notOk(args.kwargs.feedback);
+                return;
+            }
+            if (route === '/web/dataset/call_kw/mail.activity/unlink') {
+                // should not be called
```

Comment:
Shouldn't it raise/log error instead? Just asking I'm not sure, have you seen other occurrences doing useless step?

Reply:
It's a way I found to check that the `unlink` rpc is not made when `action_feedback` is called, so I add a step here and in `verifySteps` I check that it's not contained. You think it's better to force the test to fail if it goes there ?

Reply:
I think the difference between the two is that adding the step as here specifically document that the route should not be called.

My argument is that there is an infinite amount of other routes that shouldn't be called, so there should be a specific reason to list this one in particular.

If the end result is just for the test to crash, and it is crashing without this line, I would remove it.
If it's to document a specific case where `unlink` is generally called, but we want in this very specific case ensure it is not called, then I would keep it, and add a comment to explain _why_ it "should not be called".

Reply:
It cames to me because the unlink was called through activity.delete() after marking it as done (which already "delete" it from server) so the RPC was failing. So here we check if the unlink is still made or not. Like it was a bugfix check in fact .... So I would say we're in the third situation, aren't we ?

Reply:
The point about "documentation" with test is more to describe (functional) features or specific cases. 

If the RPC failing leads to the test failing, I would not add it because the bugfix becomes already covered by the test by default. For bugfix the only thing that matters is that runbot is red when there is an issue, there shouldn't be any extra code for it as long as it is red.

@alexkuhn maybe final word on this thread?

Reply:
raise error, this is a very technical test that ensures one RPC is called and the other not, so the sooner something wrong is detected the better. Also when this test fails, listing a mismatch in the verify steps is not clear enough.